src/cacheSeed/cacheSeedManager.ts

```diff
@@ -0,0 +1,87 @@
+    private readonly queueClient: QueueClient,
+    private readonly mapproxySeed: MapproxySeed
+  ) {
+    this.seedAttempts = this.config.get<number>('seedAttempts');
+    this.taskType = this.config.get<string>('queue.tilesTaskType');
+  }
+
+  public async handleCacheSeedTask(): Promise<boolean> {
+    this.logger.debug('Running search for seed jobs');
+    const tilesTask = await this.queueClient.queueHandlerForTileSeedingTasks.dequeue<ITaskParams>(this.taskType);
+    if (!tilesTask) {
+      return Boolean(tilesTask);
+    }
+
+    if (tilesTask.parameters.cacheType !== CacheType.REDIS) {
+      this.logger.error(`Unsupported cache type, only ${CacheType.REDIS} type is valid`);
+      const cacheType = tilesTask.parameters.cacheType;
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+      await this.queueClient.queueHandlerForTileSeedingTasks.reject(tilesTask.jobId, tilesTask.id, false, `Unsupported cache type ${cacheType}`);
+      await this.queueClient.queueHandlerForTileSeedingTasks.jobManagerClient.updateJob(tilesTask.jobId, {
+        status: OperationStatus.FAILED,
+        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+        reason: `Unsupported cache type ${cacheType}`,
+      });
+      return false;
+    }
+    const job = await this.queueClient.jobsClient.getJob<IJobParams, ITaskParams>(tilesTask.jobId);
+    const jobId = tilesTask.jobId;
+    const taskId = tilesTask.id;
+    const attempts = tilesTask.attempts;
+    const seeds = tilesTask.parameters.seedTasks;
+    this.logger.info({
+      msg: `Found new seed job: ${jobId}`,
+      jobId,
+      taskId,
+      productId: job.resourceId,
+      productVersion: job.version,
+      productType: job.productType,
+    });
+
+    if (attempts <= this.seedAttempts) {
+      try {
+        await this.runTask(seeds);
+        await this.queueClient.queueHandlerForTileSeedingTasks.ack(jobId, taskId);
+        await this.queueClient.queueHandlerForTileSeedingTasks.jobManagerClient.updateJob(jobId, {
+          status: OperationStatus.COMPLETED,
+          percentage: 100,
+        });
+      } catch (error) {
+        await this.queueClient.queueHandlerForTileSeedingTasks.reject(jobId, taskId, true, (error as Error).message);
+      }
+    } else {
+      this.logger.warn({ msg: `Reached to max attempts and will close task as failed`, maxAttempts: this.seedAttempts });
+      await this.queueClient.queueHandlerForTileSeedingTasks.reject(jobId, taskId, false);
+      await this.queueClient.queueHandlerForTileSeedingTasks.jobManagerClient.updateJob(jobId, { status: OperationStatus.FAILED });
+    }
+
+    return Boolean(tilesTask);
+  }
+
+  private async runTask(seedTasks: ITaskSeedOptions[]): Promise<void> {
+    for (const task of seedTasks) {
+      if (task.mode === SeedMode.SEED) {
+        await this.mapproxySeed.runSeed(task); // todo - I will consider on phase to pass other ready to run object
```

Comment:
is this "// todo" still relevant

Reply:
removed and refactor to one method of seed-clean execution